webhooks/services.py

```diff
@@ -1,14 +1,137 @@
 """Services module."""
 
+import json
+from json.decoder import JSONDecodeError
+from typing import List, Union
+import re
+
 from aioredis import Redis
 
+from models import (
+    BasicMessagesHook,
+    to_credentential_hook_model,
+    to_proof_hook_model,
+    to_connections_model,
+)
+from shared_models import TopicItem
+
 
 class Service:
     def __init__(self, redis: Redis) -> None:
         self._redis = redis
 
-    async def add_topic_entry(self, topic: str, hook: str):
+    def _is_proof(self, topic: str) -> bool:
+        proofs_identifiers = ["present_proof", "present_proof_v2_0"]
+        return topic in proofs_identifiers
+
+    def _is_credential(self, topic: str) -> bool:
+        credential_identifiers = ["issue_credential", "issue_credential_v2_0"]
+        return topic in credential_identifiers
+
+    def _proof_hook_versioned(self, item: dict) -> dict:
+        return to_proof_hook_model(item=item)
+
+    def _credential_hook_versioned(self, item: dict) -> dict:
+        return to_credentential_hook_model(item=item)
+
+    def _version_connections(self, item: dict) -> dict:
+        return to_connections_model(item=item)
+
+    def _deserialise(self, data: bytes) -> Union[str, dict]:
```

Comment:
Why is this needed? Can't we store stringified json in redis so we don't have to transform python dict values to json values?

Reply:
As below. Yes, possible in principle